voxblox/include/voxblox/core/layer_inl.h

```diff
@@ -165,7 +165,8 @@ bool Layer<VoxelType>::addBlockFromProto(const BlockProto& block_proto,
         if (it == block_map_.end()) {
           block_map_[block_index] = block_ptr;
         } else {
-          it->second->mergeBlock(*block_ptr);
+          CHECK(it->second->mergeBlock(*block_ptr)) << "Block merging not "
```

Comment:
I rather have the crash coming from the function that is not implemented yet. Using the return type as a indicator whether or not the function is implemented is wrong, for some voxel type the return type might be used to communicate some informatio that you can't or shouldn't merge the two blocks.